src/hooks/constants/lib/fetchConstants.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { ApiPromise } from '@polkadot/api';
+import { Constants } from '../../../generated/graphql';
+import { fetchConstants } from './fetchConstants';
+import {
+  getMockApiPromise,
+  mockedRepayFee,
+} from '../../polkadotJs/tests/mockUsePolkadotJsContext';
+
+describe('fetch chain constants', () => {
+  let mockApiInstance: ApiPromise;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockApiInstance = getMockApiPromise();
+  });
+
+  it('can retrieve lbp.getRepayFee constant', () => {
+    mockApiInstance.createType = jest
+      .fn()
+      .mockImplementation(() => ({ toHuman: jest.fn(() => mockedRepayFee) }));
+
+    const chainConstants: Constants = fetchConstants(mockApiInstance);
+
+    expect(chainConstants.lbp.repayFee).toEqual({
+      numerator: '2',
+      denominator: '10',
+    });
+  });
+});
```

Comment:
Analysis:  
all stuff that needs to be set up to test the fetch should be modularized here and not polluting other functionality.

solution:
the comments are pretty clear about what needs to be moved where and why.  I'll implement what they say.